src/test/java/com/spencerwi/either/EitherFlatMapTest.java

```diff
@@ -0,0 +1,94 @@
+package com.spencerwi.either;
+
+import org.junit.jupiter.api.Test;
+
+class EitherFlatMapTest {
+
+    @Test
+    void eitherFlatMapExample() {
+
+        Either<Error, Final> either = eitherStep1(new Request())
+                .flatMapRight(this::eitherStep2)
+                .mapRight(this::step3);
```

Comment:
Would it be possible actually to "inline" `eitherStep2` and `step3` as lambdas here, so that it's a little more obvious at a glance what the sort of "signature" is? I try to have the tests serve as "documentation examples" so that people can quickly and easily see how the library behaves by reading tests.